### PR TITLE
feat: add date difference calculator with timezone support

### PR DESCRIPTION
**Pull request by imfelixyeung**
## Summary
- **New Feature**: Date Difference Calculator with timezone support
- **Testing**: Comprehensive unit tests with Vitest (16 tests passing)
- **UI Enhancement**: Whole number display with "and X days" format for weeks/months
- **Integration**: Seamlessly integrated with existing timezone store

## Key Features
- Calculate difference between two dates with timezone awareness
- Display results in multiple formats: days, weeks, months, hours, minutes, seconds
- Show whole numbers for weeks and months with remaining days
- Handle edge cases: invalid dates, end before start, empty inputs
- Full TypeScript typing with comprehensive JSDoc

## Example Output
```
Days: 69
Weeks: 9 (and 6 days)
Months (approx): 2 (and 8 days)
Hours: 1656
Minutes: 99360
Seconds: 5961600
```

## Technical Implementation
- **Core Logic**: `src/lib/converters/date-difference.tsx`
- **UI Component**: `src/components/tools/date-difference.tsx`
- **Tests**: `src/lib/converters/__tests__/date-difference.test.ts`
- **Vitest Setup**: Configuration and test scripts added
- **Integration**: Added to home page between existing tools

## Quality
- ✅ All tests passing
- ✅ Lint checks passing
- ✅ Code formatting with Prettier
- ✅ TypeScript type safety
- ✅ Follows existing project conventions

The feature is ready for review and merge!

## Summary by CodeRabbit

* **New Features**
  * Added a Date Difference Calculator with timezone-aware results, interactive date inputs, "Current" buttons, and clear error/result displays.

* **Documentation**
  * Expanded testing guidance with full command examples and updated tips for the current test runner.

* **Tests**
  * Added comprehensive tests for date calculations, timezone validation, edge cases, and test setup for UI-enabled runs.

* **Chores**
  * Added test scripts, test config and CI job; updated date-time library dependency.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/lib/converters/date-difference.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+ * @example
+ * ```typescript
+ * // Basic usage
+ * const { result, error } = calculateDateDifference("2025-10-23", "2026-02-20");
+ * console.log(result.days); // 120
+ *
+ * // With timezone
+ * const { result: result2 } = calculateDateDifference("2025-10-23", "2026-02-20", "UTC");
+ * console.log(result2.weeks); // 17.14
+ * ```
```
[the remark this answers is not in the document]
Comment (imfelixyeung) [reply src/lib/converters/date-difference.tsx]: Resolved

**Comment by imfelixyeung** [comment]
@coderabbitai generate docstrings
